lib/Dialect/OpDescription.cpp

```diff
@@ -116,215 +116,103 @@ template <> const OpDescription &OpDescription::get<BinaryOperator>() {
 #define HANDLE_USER_INST(...)
 #define HANDLE_UNARY_INST(...)
 #define HANDLE_BINARY_INST(...)
-#define HANDLE_INST(num, opcode, Class)                                        \
+#define HANDLE_INST_DESC(num, Category, opcode, Class, InstPrefix)                         \
   template <> const OpDescription &OpDescription::get<Class>() {               \
-    static const OpDescription desc{Kind::Core, Instruction::opcode};          \
+    static const OpDescription desc{Kind::Category, InstPrefix::opcode};      \
     return desc;                                                               \
   }
+#define HANDLE_INST(num, opcode, Class)                                        \
+  HANDLE_INST_DESC(num, Core, opcode, Class, Instruction)
+#define HANDLE_INTRINSIC_DESC(Class, opcode)                                   \
+  HANDLE_INST_DESC(0, Intrinsic, opcode, Class, Intrinsic)
 #include "llvm/IR/Instruction.def"
 
+#define HANDLE_INTRINSIC_DESC_ARR(Class, ...)                                  \
+  template <> const OpDescription &OpDescription::get<Class>() {               \
+    static unsigned opcodes[] = {__VA_ARGS__};                                 \
+    static const OpDescription desc{Kind::Intrinsic, opcodes};                 \
+    return desc;                                                               \
+  }
+
 // ============================================================================
 // Descriptions of intrinsic facades implemented in LLVM
 
-template <> const OpDescription &OpDescription::get<LifetimeIntrinsic>() {
-  static unsigned opcodes[] = {
-      Intrinsic::lifetime_start,
-      Intrinsic::lifetime_end,
-  };
-  static const OpDescription desc{Kind::Intrinsic, opcodes};
-  return desc;
-}
-
-template <> const OpDescription &OpDescription::get<DbgInfoIntrinsic>() {
-  static unsigned opcodes[] = {
-      Intrinsic::dbg_declare, Intrinsic::dbg_value, Intrinsic::dbg_label,
-      Intrinsic::dbg_assign,
-      // Intrinsic::dbg_addr, <-- add this back for sufficiently recent LLVM
-  };
-  static const OpDescription desc{Kind::Intrinsic, opcodes};
-  return desc;
-}
+HANDLE_INTRINSIC_DESC_ARR(LifetimeIntrinsic, Intrinsic::lifetime_start,
+                          Intrinsic::lifetime_end)
```

Comment:
The HANDLE_xxx_INST that were already there have a special meaning in connection to including Instruction.def. Please do not change them.

Keep the definition of the intrinsic-related #defines separate and below the comment.